DetectorDescription/DDCMS/src/DDDetector.cc

```diff
@@ -0,0 +1,30 @@
+#include "DetectorDescription/DDCMS/interface/DDDetector.h"
+#include "DD4hep/Detector.h"
+
+#include <iostream>
+ 
+using namespace cms;
+using namespace std;
+
+DDDetector::DDDetector(const string& tag, const string& fileName)
+  : m_tag(tag)
+{
+  m_description = &dd4hep::Detector::getInstance(tag);
```

Comment:
I looked at the DD4Hep code. I think we could easily add a function `std::unique_ptr<dd4hep::Detector> makeInstance();` which does not add the object to its internal global list and does not set `gTGeoManager` which the one held by the object. That would be the first step in making things safer to use. After adding such a function to our own version of DD4Hep, I'd propose we then also make it a pull request to their repository.

Reply:
Thanks, I’ll try it

Reply:
Isn't there an alias `Detector` for `dd4hep::Detector`? Better to use it consistently.

Reply:
it will be resolved in the next PR